code/modules/mob/living/carbon/human/human_helpers.dm

```diff
@@ -113,6 +113,89 @@
 		return
 	return held_item.GetID()
 
+/mob/living/carbon/human/proc/get_accessible_cash()
+	var/available_cash = 0
+	var/list/cash_list = get_cash_list()
+	if(!length(cash_list))
+		return 0
+	for(var/found_item in cash_list)
+		if(istype(found_item, /obj/item/holochip))
+			var/obj/item/holochip/chip_stack = found_item
+			available_cash += chip_stack.credits
+		if(istype(found_item, /obj/item/card/id))
+			var/obj/item/card/id/id_card = found_item
+			available_cash += id_card.registered_account.account_balance
+	return available_cash
+
+/mob/living/carbon/human/proc/spend_cash(var/to_spend)
+	if(!to_spend)
+		return FALSE
+	if(to_spend > get_accessible_cash()) //If we don't have enough money, early return
+		return FALSE
+	var/list/cash_list = get_cash_list()
+	if(!length(cash_list)) //Another check, just in case
+		return FALSE
+	for(var/obj/item/holochip/chip_stack in cash_list)//Holochips take priority over ID cards
+		if(chip_stack.credits >= to_spend)
+			chip_stack.spend(to_spend, TRUE)
+			return TRUE
+		else
+			var/temp_value_holder = to_spend
+			to_spend -= chip_stack.credits
+			chip_stack.spend(temp_value_holder, TRUE)
+	for(var/obj/item/card/id/id_card in cash_list)
+		var/temp_cash_holder = id_card.registered_account.account_balance
+		if(temp_cash_holder >= to_spend)
+			id_card.registered_account.adjust_money(-to_spend)
+			return TRUE
+		else
+			id_card.registered_account.adjust_money(-to_spend)
+			to_spend -= temp_cash_holder
+	return FALSE
+
+/mob/living/carbon/human/proc/get_cash_list()
```

Comment:
I think this proc should do this logic
```DM
for(var/obj/each_item in list(get_active_held_item(), get_inactive_held_item(), wear_id, belt))
    if(!each_item)
        continue // if you have nothing, skip it
    else if(istype(each_item, /obj/holochip_thing))
        // do_something
        found_list += each_item
    else if(istype(each_item, /obj/card_thing))
        // do something
        found_list += each_item
    else if(istype(each_item, /obj/wallet_thing))
        // do something
        found_list += wallet_result // this should be recursive that uses the logic above
```
also, this should be work sorta recursively because you want the same logic from wallet.